src/imguiclient/imguiapp.cpp

```diff
@@ -1410,66 +1404,77 @@ std::string ImGuiApp::create_time_string(std::chrono::time_point<std::chrono::sy
 
 void ImGuiApp::BackgroundWindow()
 {
-    // #TODO: Make the background dockable somehow. There is a example in ImGui Demo > Examples > Dockspace
+    // Configure flags for the background window
+    ImGuiWindowFlags window_flags = ImGuiWindowFlags_MenuBar | ImGuiWindowFlags_NoDocking | ImGuiWindowFlags_NoTitleBar
+        | ImGuiWindowFlags_NoCollapse | ImGuiWindowFlags_NoResize | ImGuiWindowFlags_NoMove
+        | ImGuiWindowFlags_NoBringToFrontOnFocus | ImGuiWindowFlags_NoNavFocus;
+
+    // Get the main viewport
+    const ImGuiViewport *viewport = ImGui::GetMainViewport();
+
+    // Set window position and size to cover the entire viewport
+    ImGui::SetNextWindowPos(viewport->WorkPos);
+    ImGui::SetNextWindowSize(viewport->WorkSize);
+    ImGui::SetNextWindowViewport(viewport->ID);
 
-    // clang-format off
-    constexpr ImGuiWindowFlags window_flags =
-        ImGuiWindowFlags_NoDecoration |
-        ImGuiWindowFlags_NoMove |
-        ImGuiWindowFlags_NoBackground |
-        ImGuiWindowFlags_MenuBar |
-        ImGuiWindowFlags_NoBringToFrontOnFocus |
-        ImGuiWindowFlags_NoDocking;
-    // clang-format on
+    ImGui::PushStyleVar(ImGuiStyleVar_WindowRounding, 0.0f);
+    ImGui::PushStyleVar(ImGuiStyleVar_WindowBorderSize, 0.0f);
 
+    // Push zero padding to maximize space
+    ImGui::PushStyleVar(ImGuiStyleVar_WindowPadding, ImVec2(0.0f, 0.0f));
+
+    // Begin the background window
     bool window_open = true;
-    ImScoped::Window window("main", &window_open, window_flags);
-    if (window.IsContentVisible)
+    ImGui::Begin("DockSpace", &window_open, window_flags);
```

Comment:
Why was `ImScoped::Window window("main", &window_open, window_flags);` replaced with this `ImGui::Begin("DockSpace", &window_open, window_flags);`? It no longer checks the return and the ImScoped looked just fine.